pkg/resourcegroup/controllers/resourcegroup/resourcegroup_controller.go

```diff
@@ -205,20 +193,29 @@ func (r *reconciler) endReconcilingStatus(
 	}()
 	select {
 	case <-finish:
-		newStatus.ObservedGeneration = generation
-		// Update conditions, if they've changed
-		newStatus.Conditions = updateCondition(newStatus.Conditions,
-			newReconcilingCondition(v1alpha1.FalseConditionStatus, FinishReconciling, finishReconcilingMsg))
 		newStatus.Conditions = updateCondition(newStatus.Conditions,
 			aggregateResourceStatuses(newStatus.ResourceStatuses))
 	case <-time.After(reconcileTimeout):
-		// Update conditions, if they've changed
-		newStatus.Conditions = updateCondition(newStatus.Conditions,
-			newReconcilingCondition(v1alpha1.FalseConditionStatus, ExceedTimeout, exceedTimeoutMsg))
 		newStatus.Conditions = updateCondition(newStatus.Conditions,
 			newStalledCondition(v1alpha1.TrueConditionStatus, ExceedTimeout, exceedTimeoutMsg))
 	}
 
+	// Updating the ObservedGeneration is probably a no-op, because the applier
+	// should have updated it already when it updated the spec & status.
+	// But update it here just for completeness.
+	// It's important to note that because there's only one ObservedGeneration
+	// field, it cannot be used to confirm that this controller has completed.
+	// it can only be used to confirm that the applier has completed its update.
+	newStatus.ObservedGeneration = generation
```

Comment:
It looks like this is a behavioral change - before it did not set the observedGeneration in the case of `reconcileTimeout` exceeded. Rather, it only set the observedGeneration after successfully computing the status.

Is this an intentional change?

Reply:
Your right. Maybe I should move it out to its own PR. 

Ideally it should be a no-op, assuming that the inventory client already updated the ObservedGeneration when it updated the status. However, I think there's actually a bug in the RG inventory client that updates this to the generation before it updates the spec, causing it to be out of date.

Reply:
Reverted this change.